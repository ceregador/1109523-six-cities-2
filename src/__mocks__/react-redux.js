export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
