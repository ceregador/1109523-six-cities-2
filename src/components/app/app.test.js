import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import App from './app.jsx';
import MainPage from '../mainPage/main-page.jsx';
import SignIn from '../signIn/sign-in.jsx';
import Renderer from 'react-test-renderer';

jest.mock(`../mainPage/main-page.jsx`);
jest.mock(`../signIn/sign-in.jsx`);
jest.mock(`react-redux`);
jest.mock(`../../selectors/selector`);

it(`renders SignIn page if not authorized and connects correctly`, () => {
  const tree = Renderer
      .create(
          <App
            isAuthorized={false}
            tryToAuthorize={() => undefined}
          />)
      .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({});
  expect(mappedProps).toHaveProperty(`isAuthorized`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`tryToAuthorize`);

  expect(connectAdvanced).toHaveBeenCalledWith(App);

  expect(SignIn).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});

it(`renders MainPage if authorized and connects correctly`, () => {
  const tree = Renderer
        .create(
            <App
              isAuthorized={true}
              tryToAuthorize={() => undefined}
            />)
        .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({});
  expect(mappedProps).toHaveProperty(`isAuthorized`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`tryToAuthorize`);

  expect(connectAdvanced).toHaveBeenCalledWith(App);

  expect(MainPage).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
