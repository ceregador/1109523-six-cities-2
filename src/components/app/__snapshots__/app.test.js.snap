// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage if authorized and connects correctly 1`] = `null`;

exports[`renders SignIn page if not authorized and connects correctly 1`] = `null`;
